python/lsst/sims/catUtils/utils/alertGenerator.py

```diff
@@ -0,0 +1,1475 @@
+                photometrically_valid[i_obj] = 1
+
+        photometrically_valid = np.where(photometrically_valid>=0)
+
+        if 'properMotionRa'in column_query:
+            pmra = chunk['properMotionRa'][photometrically_valid]
+            pmdec = chunk['properMotionDec'][photometrically_valid]
+            px = chunk['parallax'][photometrically_valid]
+            vrad = chunk['radialVelocity'][photometrically_valid]
+        else:
+            pmra = None
+            pmdec = None
+            px = None
+            vrad = None
+
+        ###################################################################
+        # Figure out which sources actually land on an LSST detector during
+        # the observations in question
+        #
+        t_before_chip_name = time.time()
+        chip_name_dict = {}
+
+        # time_arr will keep track of which objects appear in which observations;
+        # 1 means the object appears; -1 means it does not
+        time_arr_transpose = -1*np.ones((len(obs_valid_dex), len(chunk['raJ2000'])),
+                                        dtype=int)
+
+        for i_obs, obs_dex in enumerate(obs_valid_dex):
+            obs = self._obs_list[obs_dex]
+            chip_name_list = np.array([None]*n_raw_obj)
+            xPup_list = np.zeros(n_raw_obj, dtype=float)
+            yPup_list = np.zeros(n_raw_obj, dtype=float)
+            chip_int_arr = -1*np.ones(len(chip_name_list), dtype=int)
+
+            if len(photometrically_valid[0])>0:
+                xPup_list_val, yPup_list_val = _pupilCoordsFromRaDec(chunk['raJ2000'][photometrically_valid],
+                                                                     chunk['decJ2000'][photometrically_valid],
+                                                                     pm_ra=pmra, pm_dec=pmdec,
+                                                                     parallax=px, v_rad=vrad,
+                                                                     obs_metadata=obs)
+
+                xPup_list[photometrically_valid] = xPup_list_val
+                yPup_list[photometrically_valid] = yPup_list_val
+
+                chip_name_list[photometrically_valid] = chipNameFromPupilCoordsLSST(xPup_list_val,
+                                                                                    yPup_list_val)
+
+                for i_chip, name in enumerate(chip_name_list):
+                    if name is not None:
+                        chip_int_arr[i_chip] = 1
+
+            valid_obj = np.where(chip_int_arr>0)
+            time_arr_transpose[i_obs][valid_obj] = 1
+
+            chip_name_dict[i_obs] = (chip_name_list,
+                                     xPup_list,
+                                     yPup_list,
+                                     valid_obj)
+
+        time_arr = time_arr_transpose.transpose()
+        assert len(chip_name_dict) == len(obs_valid_dex)
+
+        ######################################################
```

Comment:
Actually, that won't work.  `_filter_on_photometry_then_chip_name(chunk)` sets `photometry_catalog.current_chunk` equal to `chunk`, so that this call to `photometry_catalog.column_by_name('decICRS')` returns the correct subset of objects.

Having written that, I can see how that is very opaque.

I wanted to separate the `_filter_on...` code, though, so I could easily tweak and replace it, since that is the principal bottleneck in this process.

Would it be clearer if `_filter_on_photometry_then_chip_name` returned `photometry_catalog`, so that it was obvious that `photometry_catalog` was affected by the function call?

Reply:
Oh, no. This was me being opaque, haha. I meant the commented lines below seemed like they were supposed to describe what was happening at the start of the function.

Reply:
And maybe the last couple commented lines here didn't apply to this version of the function and were accidentally left in from editing the other version.

Reply:
Ah.  Right.  I'll move the comments